Flaky io.camunda.zeebe.broker.exporter.stream.ExporterDirectorTest#shouldConfigureAllExportersProperlyOnStart

**Summary**

- How often does the test fail? 1
- Does it block your work? No
- Do we suspect that it is a real failure? No

**Failures**


```
Error:  Tests run: 18, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 11.98 s <<< FAILURE! -- in io.camunda.zeebe.broker.exporter.stream.ExporterDirectorTest
Error:  io.camunda.zeebe.broker.exporter.stream.ExporterDirectorTest.shouldConfigureAllExportersProperlyOnStart -- Time elapsed: 0.394 s <<< FAILURE!
java.lang.AssertionError: 

Expecting actual not to be null
	at io.camunda.zeebe.broker.exporter.stream.ExporterDirectorTest.lambda$shouldConfigureAllExportersProperlyOnStart$8(ExporterDirectorTest.java:226)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at io.camunda.zeebe.broker.exporter.stream.ExporterDirectorTest.shouldConfigureAllExportersProperlyOnStart(ExporterDirectorTest.java:224)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
```


**Logs**

[Test results for unit tests.zip](https://github.com/camunda/zeebe/files/12427550/Test.results.for.unit.tests.1.zip)

--- a/broker/src/test/java/io/camunda/zeebe/broker/exporter/stream/ExporterDirectorTest.java
+++ b/broker/src/test/java/io/camunda/zeebe/broker/exporter/stream/ExporterDirectorTest.java
@@ -55,7 +55,9 @@ public final class ExporterDirectorTest {
   private static final String EXPORTER_ID_1 = "exporter-1";
   private static final String EXPORTER_ID_2 = "exporter-2";
 
-  private static final VerificationWithTimeout TIMEOUT = timeout(5_000);
+  private static final int TIMEOUT_MILLIS = 5_000;
+  private static final VerificationWithTimeout TIMEOUT = timeout(TIMEOUT_MILLIS);
+
   @Rule public final ExporterRule rule = ExporterRule.activeExporter();
   private final List<ControlledTestExporter> exporters = new ArrayList<>();
   private final List<ExporterDescriptor> exporterDescriptors = new ArrayList<>();
@@ -213,11 +215,14 @@ public final class ExporterDirectorTest {
   }
 
   @Test
-  public void shouldConfigureAllExportersProperlyOnStart() {
+  public void shouldConfigureAllExportersProperlyOnStart() throws InterruptedException {
     // when
+    final CountDownLatch latch = new CountDownLatch(exporters.size());
+    exporters.forEach(exporter -> exporter.onOpen(c -> latch.countDown()));
     startExporterDirector(exporterDescriptors);
 
     // then
+    assertThat(latch.await(TIMEOUT_MILLIS, TimeUnit.MILLISECONDS)).isTrue();
     verify(exporters.get(0), TIMEOUT).open(any());
     verify(exporters.get(1), TIMEOUT).open(any());
 

